File: 프로그래머스/1/147355. 크기가 작은 부분문자열/크기가 작은 부분문자열.java
class Solution {
    public int solution(String t, String p) {
        int answer = 0;
        for (int i = 0; i < t.length()-p.length()+1; i++) {
            long n = Long.parseLong(t.substring(i,p.length()+i));
            if(Long.parseLong(p)>=n){
                answer ++;
            }
        }
        return answer;
    }
}

File: 프로그래머스/1/147355. 크기가 작은 부분문자열/크기가 작은 부분문자열.kt
class Solution {
    fun solution(t: String, p: String): Int {
        var answer: Int = 0

        for (i in 0 until t.length - p.length + 1) {
            var subString = t.substring(i, p.length + i)

            if (p >= subString) {
                answer += 1
            }
        }
        return answer
    }
}